diff --git a/hack/generate-crd-deepcopy.sh b/hack/generate-crd-deepcopy.sh
index 6778189c..60d53666 100755
--- a/hack/generate-crd-deepcopy.sh
+++ b/hack/generate-crd-deepcopy.sh
@@ -39,8 +39,10 @@ readonly HEADER=$(mktemp)
 boilerplate > "${HEADER}"
 
 exec echo "controller-gen version: "
-exec go run sigs.k8s.io/controller-tools/cmd/controller-gen --version
+#exec go run sigs.k8s.io/controller-tools/cmd/controller-gen --version
+exec /home/tsaarni/work/controller-tools/controller-gen --version
 
-exec go run sigs.k8s.io/controller-tools/cmd/controller-gen \
+#exec go run sigs.k8s.io/controller-tools/cmd/controller-gen \
+exec /home/tsaarni/work/controller-tools/controller-gen \
     "object:headerFile=${HEADER}" \
     "paths=${PATHS}"
diff --git a/hack/generate-crd-yaml.sh b/hack/generate-crd-yaml.sh
index 4ffc3a32..954f3e08 100755
--- a/hack/generate-crd-yaml.sh
+++ b/hack/generate-crd-yaml.sh
@@ -16,12 +16,14 @@ trap 'rm -rf "$TEMPDIR"; exit' 0 1 2 15
 cd "${REPO}"
 
 echo "controller-gen version: "
-go run sigs.k8s.io/controller-tools/cmd/controller-gen --version
+#go run sigs.k8s.io/controller-tools/cmd/controller-gen --version
+/home/tsaarni/work/controller-tools/controller-gen --version
 
 # Controller-gen seems to use an unstable sort for the order of output of the CRDs
 # so, output them to separate files, then concatenate those files.
 # That should give a stable sort.
-go run sigs.k8s.io/controller-tools/cmd/controller-gen \
+#go run sigs.k8s.io/controller-tools/cmd/controller-gen \
+/home/tsaarni/work/controller-tools/controller-gen \
   crd:crdVersions=v1 "paths=${PATHS}" "output:dir=${TEMPDIR}"
 
 # Explicitly add "preserveUnknownFields: false" to CRD specs since any CRDs created
